### DISCUSSION TITLE
Style command

### DISCUSSION THREAD

**Discussion by Bharat0112** [okaybenji/text-engine#32] [Q&A] [answered]
I have added the following code for stylesheet changing from the game's demo-disk file. It changes stylesheet by typing stylex. It circles among 3 already given stylesheets. Is there any way to add stylex as a command that takes filename as an argument, so I can add multiple styles. For example 'stylex classic' will change the style to classic.css.

```
let stylex = () => {
  const selectStylesheet = filename => document.getElementById('styles').setAttribute('href', filename);
  const currentStylesheet = document.getElementById('styles').getAttribute('href');
  
  let newName;
  if (currentStylesheet.includes('modern')) {
    newName = 'retro';
  } else if (currentStylesheet.includes('retro')) {
    newName = 'classic';
  } else {
    newName = 'modern';
  }
  
  println(`You changed the stylesheet to ${newName.toUpperCase()}.css.`);
  selectStylesheet(`styles/${newName}.css`);
};
```

**Comment by okaybenji** [comment, marked as answer]
Yeah, that's no problem. Change your function to accept `newName` as an argument like this:

```
let stylex = (newName) => {
  const selectStylesheet = filename => document.getElementById('styles').setAttribute('href', filename);
  println(`You changed the stylesheet to ${newName.toUpperCase()}.css.`);
  selectStylesheet(`styles/${newName}.css`);
};
```

Now you can call the function, passing the filename as a string:

```
stylex('modern');
```

Finally, you'll need to update your game's commands. Since this function accepts a single argument, you'll need to override `commands[1]` like this:

```
commands[1] = Object.assign(commands[1], {stylex});
```

Check out the [Commands](https://github.com/okaybenji/text-engine#commands) section of the documentation for more info.

**Reply by Bharat0112** [reply]
Hello, 
I put the fuction 

`let stylex = (newName) => {
  const selectStylesheet = filename => document.getElementById('styles').setAttribute('href', filename);
  println(`You changed the stylesheet to ${newName.toUpperCase()}.css.`);
  selectStylesheet(`styles/${newName}.css`);
};`

But I have no idea what to do with 
stylex('modern'); and 
commands[1] = Object.assign(commands[1], {stylex});

**Comment by okaybenji** [comment]
> But I have no idea what to do with stylex('modern'); and commands[1] = Object.assign(commands[1], {stylex});

`stylex('modern')` is just an example I included to demonstrate the way the function would work, but you don't need to use it.

`commands[1] = Object.assign(commands[1], {stylex});` is code you'll need to include alongside your stylex function definition in order to make it so typing `stylex modern` in the game will change the style to modern.

**Reply by Bharat0112** [reply]
Ok, in index.js file, I have put `commands[1] = Object.assign(commands[1], {stylex});` on line number 759 just above `const [command, ...args] = values;` and stylex command works. 
Thank you very much.

**Reply by okaybenji** [reply]
Glad to hear it! Have fun.
